### DISCUSSION TITLE
return empty array if pagination cursor is not matched where condition

### DISCUSSION THREAD

**Discussion by hiphapis** [prisma/prisma#21824] [Q&A]
### Bug description

```
Post
---------
id | status
----------
1 | active
2 | active
3 | active
4 | active
5 | deleted <- deleted
6 | active
7 | deleted <- deleted
8 | active
```

```ts
await this.prisma.post.findMany({
  cursor: { id: 4 },
  skip: 1,
  where: { status: 'active' }
})
```
It works. No problem!
But assign deleted post id then return empty array
```ts
await this.prisma.post.findMany({
  cursor: { id: 5 }, // pick deleted row
  skip: 1,
  where: { status: 'active' } // find active
})
// return []
```

I did few experiment, the result is..
conflict cursor id & where condition, then return empty array

if i change status to `deleted`. it works
```ts
await this.prisma.post.findMany({
  cursor: { id: 5 },
  skip: 1,
  where: { status: 'deleted' } // change status to coursor row status, then works
})
```

### How to reproduce

I wrote to `Bug description` section

### Expected behavior

```ts
await this.prisma.post.findMany({
  cursor: { id: 5 },
  skip: 1,
  where: { status: 'active' }
})
```
then expected id 6, 8 rows (7 row is deleted. so except it)

### Prisma information

```prisma
// Add your schema.prisma
model Post {
  id             Int       @id @default(autoincrement())
  user_id        Int // @map("user_id")
  board_id       Int? // @map("board_id")
  status         String?   @db.VarChar(50)
  views_count    Int? // @map("view_count")
  comments_count Int       @default(0) @db.UnsignedInt
  likes_count    Int       @default(0) @db.UnsignedInt
  reports_count  Int       @default(0) @db.UnsignedInt
  related_data   Json?
  delete_time    DateTime? @db.DateTime(0) // @map("delete_time")
  create_time    DateTime? @db.DateTime(0) // @map("create_time")
  update_time    DateTime? @db.DateTime(0) // @map("update_time")
  content        String?   @db.Text
  title          String?   @db.VarChar(500)

  user     User?     @relation(fields: [user_id], references: [id])
  comments Comment[]
  likes    Like[]
  images   Image[]

  @@index([user_id], map: "ix_community_post_user_id")
}

```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: macOS
- Database: MySQL
- Node.js version: 20.9.0

### Prisma Version

```
prisma                  : 5.5.2
@prisma/client          : 5.5.2
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.5.1-1.aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a
Default Engines Hash    : aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a
Studio                  : 0.494.0
Preview Features        : multiSchema
```

**Comment by hiphapis** [comment]
I wanna show the Prisma pagination code. but i couldn't found code
Anybody know this?

**Comment by ludralph** [comment]
Hi @hiphapis 👋 

Thank you for raising this question.

When using Prisma ORM for cursor-based pagination, if you provide a cursor that does not exist in the table, Prisma will return null. This behavior occurs even if there are records before and after the specified cursor position in the table. Prisma does not attempt to locate adjacent values to the nonexistent cursor. See this section of the [documentation](https://www.prisma.io/docs/orm/prisma-client/queries/pagination#what-happens-if-the-cursor-value-does-not-exist).

We do have an open [issue](https://github.com/prisma/prisma/issues/18153) to improve this behaviour. You can take a look at the issue to see the recommendations for improving the behavior of cursor based pagination when the id does not exist.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
